### PR TITLE
ONNX: Gemm operator

### PR DESCRIPTION
**Pull request by aclex**
This one is also sometimes happened in the models as a result of certain matmul export.

### CODE REVIEW

**Comment by aclex** [comment]
Well, the request seems to include other requests' material, so could you please review them in sequence, BatchNorm, Global Pooling and this one, so that I could rebase it in time.

**Comment by aclex** [comment]
Edit: rebased against current master, so no dependency on other requests anymore.

**Comment by sunnycase** [comment]
Looks like the test didn't pass.

**Comment by aclex** [comment]
Yes, seems that PyTorch ONNX export fails again here. Will look at it. Where can I see the PyTorch version in this container, by the way?

**Comment by sunnycase** [comment]
@aclex At `azure-pipelines.yml`
https://github.com/kendryte/nncase/blob/master/azure-pipelines.yml#L39
https://github.com/kendryte/nncase/blob/master/azure-pipelines.yml#L92
https://github.com/kendryte/nncase/blob/master/azure-pipelines.yml#L155

**Comment by aclex** [comment]
@sunnycase thank you for pointing out! Will try to use 1.4.0-compliant PyTorch code or maybe pure ONNX code instead. I'm still in two minds on what's the better for tests in general: on the one hand pure ONNX export has direct clarity and better compatibility, on the other hand it's usually desired to also test PyTorch-generated ONNX code, as it's going to be the main use case for ONNX backend.

**Comment by sunnycase** [comment]
@aclex IMO using PyTorch-generated ONNX is more pratical and valuable.

**Comment by aclex** [comment]
@sunnycase tried to finally fix it with the last dimension of the second matmul operand reduction, but it's pretty much looks like random crash somewhere inside the PyTorch, as it's even not starting to generate the code for Gemm operator itself, so seems to be unrelated to any particular ONNX operator.

**Comment by aclex** [comment]
Well, it keeps crashing in spite of the fixes, I can't reproduce it on my PyTorch 1.5 and figure out what the problem is, so the only solution obvious to me is to bump PyTorch version in the test configuration to 1.5 or higher. This is generally a good measure, as ONNX stack evolves there rapidly as well, so certain bugs might be already fixed.

**Comment by aclex** [comment]
Many thanks for the merge!

@sunnycase I've noticed, there's some "small refactoring" (tm) under way. Are there any plans to keep ONNX backend after the update? I kind of plan to debug the ONNX import some more, so being in question whether to do it sooner on the current master or later, after the changes are done.

**Comment by sunnycase** [comment]
@aclex nncase is now under heavy refactoring in [features/upgrade](https://github.com/kendryte/nncase/tree/features/upgrade) branch and ONNX importer will be extended.
There is a working draft about ONNX importer and you can take a look at it https://github.com/kendryte/nncase/pull/215.

**Comment by aclex** [comment]
@sunnycase oh, nice, thanks for the information, will definitely look into it!
